merlion/models/anomaly/dagmm.py

```diff
@@ -203,6 +206,46 @@ def train(
         )
         return train_scores
 
+    def train_multiple(
+        self, train_data: List[TimeSeries], anomaly_labels: List[TimeSeries] = None,
+        train_config=None, post_rule_train_config=None
+    ) -> List[TimeSeries]:
+        """
+        Trains the anomaly detector (unsupervised) and its post-rule
+        (supervised, if labels are given) on the input multiple time series.
+
+        :param train_data: a list of `TimeSeries` of metric values to train the model.
+        :param anomaly_labels: a list of `TimeSeries` indicating which timestamps are
+            anomalous. Optional.
+        :param train_config: Additional training config dict:
+            "n_epochs": int indicating how many times the model must be trained on the `train_data`
+            "shuffle": bool indicating if the `train_data` should be shuffled before every epoch
+        :param post_rule_train_config: The config to use for training the
+            model's post-rule. The model's default post-rule train config is
+            used if none is supplied here.
+
+        :return: A list of `TimeSeries` of the model's anomaly scores on the training
+            data with each element corresponds to time series from `train_data`.
+        """
+        if anomaly_labels is not None:
+            assert len(train_data) == len(anomaly_labels)
+        else:
+            anomaly_labels = [None] * len(train_data)
+        n_epochs = train_config.pop("n_epochs", 1)
+        shuffle = train_config.pop("shuffle", False)
+        train_scores_list = []
+        for _ in range(n_epochs):
+            if shuffle:
+                random.shuffle(train_data)
+            for train_series, anomaly_series in zip(train_data, anomaly_labels):
+                train_scores_list.append(
+                    self.train(
+                        train_data=train_series, anomaly_labels=anomaly_series,
+                        train_config=train_config, post_rule_train_config=post_rule_train_config
+                    )
+                )
+        return train_scores_list
```

Comment:
There is a potential issue here, where the post-rule (calibrator and threshold) is trained individually on each time series. I think this is fine for the time being, but can you add a `#FIXME` comment here saying that the post-rule needs to be re-trained on the `train_scores` from all the models?

Reply:
Yes, I was not sure how this incremental training would affect the `post_rule`...

> on the train_scores from all the models
Do you mean from all the epochs?

Reply:
Yes, from all the epochs.